tests/robustness/engine/action.go

```diff
@@ -60,23 +52,60 @@ func (e *Engine) ExecAction(ctx context.Context, actionKey ActionKey, opts map[s
 	// If error was just a no-op, don't bother logging the action
 	switch {
 	case errors.Is(err, robustness.ErrNoOp):
-		e.RunStats.IncrementNoOpCount()
-		e.CumulativeStats.IncrementNoOpCount()
+		e.RunStats.NoOpCount++
+		e.CumulativeStats.NoOpCount++
 
 		return out, err
 
 	case err != nil:
 		log.Printf("error=%q", err.Error())
 	}
 
-	e.RunStats.RecordAction(actionKey, st, err)
-	e.CumulativeStats.RecordAction(actionKey, st, err)
-
-	e.EngineLog.AddCompleted(logEntry, err)
+	e.updatePerActionStats(actionKey, st, err)
+	e.logCompleted(logEntry, err)
 
 	return out, err
 }
 
+func (e *Engine) incrActionCountAndLog(actionKey ActionKey) {
```

Comment:
Both these functions should probably be moved to the `stats.go` file. Perhaps also use "stats" as a prefix in the name.